drivers/led/lp5562.c

```diff
@@ -945,16 +945,30 @@ static const struct led_driver_api lp5562_led_api = {
 	.off = lp5562_led_off,
 };
 
-#define LP5562_DEFINE(id)						\
-	static const struct lp5562_config lp5562_config_##id = {	\
-		.bus = I2C_DT_SPEC_INST_GET(id),			\
-	};								\
-									\
-	struct lp5562_data lp5562_data_##id;				\
-	DEVICE_DT_INST_DEFINE(id, &lp5562_led_init, NULL,		\
-			&lp5562_data_##id,				\
-			&lp5562_config_##id, POST_KERNEL,		\
-			CONFIG_LED_INIT_PRIORITY,			\
-			&lp5562_led_api);				\
+#define LP5562_DEFINE(id)                                                                          \
+	BUILD_ASSERT(DT_INST_PROP_OR(id, current_blue_ua, LP5562_DEFAULT_LED_CURRENT_UA) <=        \
+			     LP5562_MAX_LED_CURRENT_UA,                                            \
+		     "Blue channel current must be between 0 and 25.5mA.");                        \
+	BUILD_ASSERT(DT_INST_PROP_OR(id, current_green_ua, LP5562_DEFAULT_LED_CURRENT_UA) <=       \
+			     LP5562_MAX_LED_CURRENT_UA,                                            \
+		     "Green channel current must be between 0 and 25.5mA.");                       \
+	BUILD_ASSERT(DT_INST_PROP_OR(id, current_red_ua, LP5562_DEFAULT_LED_CURRENT_UA) <=         \
+			     LP5562_MAX_LED_CURRENT_UA,                                            \
+		     "Red channel current must be between 0 and 25.5mA.");                         \
+	BUILD_ASSERT(DT_INST_PROP_OR(id, current_white_ua, LP5562_DEFAULT_LED_CURRENT_UA) <=       \
+			     LP5562_MAX_LED_CURRENT_UA,                                            \
+		     "White channel current must be between 0 and 25.5mA.");                       \
```

Comment:
See earlier comment regarding handling default and max values in devicetree binding.

Reply:
ya good point :) But we still would want the default to come from the binding right?